src/transmitter/index.ts

```diff
@@ -114,3 +112,15 @@ async function retryRequest(verb: NeedleHttpVerbs, url: string, payload: object)
 
   return {response, attempt};
 }
+
+function shouldRetryRequest(err: IRequestError, networkErrorMessages: string[], stillHaveRetries: boolean): boolean {
+  if (!stillHaveRetries) {
```

Comment:
that's not a real review, just one comment about this function :) I don't like `else` when there is no need. just remove all the `else`:
```
if (something) {
  return something;
}

if (something else) {
  return something else;
}

return false;
```

Reply:
Thanks @shaimendel! Definitely it's a good a good comment to get the code style!